src/master-playlist-controller.js

```diff
@@ -487,27 +547,23 @@ export class MasterPlaylistController extends videojs.EventTarget {
   setupSegmentLoaderListeners_() {
     this.mainSegmentLoader_.on('bandwidthupdate', () => {
       const nextPlaylist = this.selectPlaylist();
-      const currentPlaylist = this.masterPlaylistLoader_.media();
-      const buffered = this.tech_.buffered();
-      const forwardBuffer = buffered.length ?
-        buffered.end(buffered.length - 1) - this.tech_.currentTime() : 0;
-
-      const bufferLowWaterLine = this.bufferLowWaterLine();
-
-      if (shouldSwitchToMedia({
-        currentPlaylist,
-        nextPlaylist,
-        forwardBuffer,
-        bufferLowWaterLine,
-        duration: this.duration(),
-        log: this.logger_
-      })) {
+
+      if (this.shouldSwitchToMedia_(nextPlaylist)) {
         this.masterPlaylistLoader_.media(nextPlaylist);
       }
 
       this.tech_.trigger('bandwidthupdate');
     });
+
     this.mainSegmentLoader_.on('progress', () => {
+      if (this.bufferWaterLineSelector) {
```

Comment:
Sometimes, especially with low bandwidth, or when bandwidth gets slower, we do not switch fast enough by waiting for an entire segment to download.

Reply:
I debated using this on without the flag, but I think we want to keep the current playlist switching intact wherever possible, so I did not do it.